go/fixchain/url_cache.go

```diff
@@ -0,0 +1,64 @@
+package fixchain
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+)
+
+type urlCache struct {
+	client *http.Client
+	cache  map[string][]byte
+	// counters may not be totally accurate due to non-atomicity
+	hit       uint
+	miss      uint
+	errors    uint
+	badStatus uint
+	readFail  uint
+}
+
+func (u *urlCache) getURL(url string) ([]byte, error) {
+	r, ok := u.cache[url]
+	if ok {
+		u.hit++
+		return r, nil
+	}
+	c, err := u.client.Get(url)
+	if err != nil {
+		u.errors++
+		return nil, err
+	}
+	defer c.Body.Close()
+	if c.StatusCode != 200 {
```

Comment:
Errors are classified as permanent or temporary (these are HTTP errors I am talking about). Cache permanent errors...

OK to leave as a TODO.